aspose/src/main/java/nl/nn/adapterframework/extensions/aspose/pipe/PdfPipe.java

```diff
@@ -154,8 +158,13 @@ public PipeRunResult doPipe(Message input, PipeLineSession session) throws PipeR
 					XmlBuilder main = new XmlBuilder("main");
 					cisConversionResult.buildXmlFromResult(main, true);
 
-					session.put("documents", main.toXML());
-					return new PipeRunResult(getSuccessForward(), main.toXML());
+					Message message = new Message(main.toXML());
+					populateContext(cisConversionResult, message, 0);
+
+					session.put("documents", message);
```

Comment:
It would be nice to have this session-key configurable.

Reply:
Actually on further review why not use the already existing configuration-option `mainDocumentSessionKey` which is used to store the result of combining PDFs?